lib/tasks/batch.rake

```diff
@@ -38,9 +39,20 @@ namespace :batch do
     return json
   end
 
-  def ingest(json, batch_dir, investigation_id, mode)
-    report = File.open(INGEST_REPORTS+"/investigation"+investigation_id+"-"+Time.now.strftime('%Y-%m-%d_%H-%M-%S')+".csv", 'a+')
-    report.puts "#{investigation_id}, #{Time.now}"
+  
+  def generate_ingest_report(noid_list, investigation_id)
+    report = File.open(INGEST_REPORTS+"/investigation_"+investigation_id+"_"+Time.now.strftime('%Y-%m-%d_%H-%M-%S')+".csv", 'a+')
+    report.puts "url|citation|title|doi|noid"
+    report.puts "investigation_id: #{investigation_id}, ingest time:  #{Time.now}"
+    noid_list.each do |noid|
+      gf = GenericFile.find(noid)
+      report.puts "#{Rails.application.routes.url_helpers.generic_file_url(gf.id)}|#{gf.is_version_of}|#{gf.title}|#{gf.doi_permanent_url}|#{gf.id}" 
+    end
+    report.close
```

Comment:
If you do it this way:

```ruby
File.open("#{INGEST_REPORTS}/investigation_#{investigation_id}_#{time}.csv", 'a+') do |report|
  report.puts "url|citation|title|doi|noid"
  report.puts "investigation_id: #{investigation_id}, ingest time:  #{Time.now}"
  noid_list.each do |noid|
     gf = GenericFile.find(noid)
     report.puts "#{Rails.application.routes.url_helpers.generic_file_url(gf.id)}|#{gf.is_version_of}|#{gf.title}|#{gf.doi_permanent_url}|#{gf.id}" 
  end
end
```

Then report is automatically closed at the end of the `File.open ... do` block, even if an exception is thrown.